src/engine/strat_engine/crypt/handle/v2.rs

```diff
@@ -0,0 +1,1252 @@
+            .context_handle()
+            .load::<()>(Some(EncryptionFormat::Luks2), None)
+        {
+            return Err(wipe_fallback(physical_path, StratisError::from(e)));
+        }
+
+        Ok(())
+    }
+
+    fn initialize_with_err(
+        device: &mut CryptDevice,
+        physical_path: &Path,
+        pool_uuid: PoolUuid,
+        encryption_info: &EncryptionInfo,
+        luks2_params: Option<&CryptParamsLuks2>,
+    ) -> StratisResult<()> {
+        let mut luks2_params_ref: Option<CryptParamsLuks2Ref<'_>> =
+            luks2_params.map(|lp| lp.try_into()).transpose()?;
+
+        log_on_failure!(
+            device.context_handle().format::<CryptParamsLuks2Ref<'_>>(
+                EncryptionFormat::Luks2,
+                ("aes", "xts-plain64"),
+                None,
+                libcryptsetup_rs::Either::Right(STRATIS_MEK_SIZE),
+                luks2_params_ref.as_mut()
+            ),
+            "Failed to format device {} with LUKS2 header",
+            physical_path.display()
+        );
+
+        match encryption_info {
+            EncryptionInfo::Both(kd, (pin, config)) => {
+                let mut parsed_config = config.clone();
+                let y = interpret_clevis_config(pin, &mut parsed_config)?;
+                Self::initialize_with_both(device, physical_path, kd, (pin, &parsed_config, y))?
+            }
+            EncryptionInfo::KeyDesc(kd) => Self::initialize_with_keyring(device, kd)?,
+            EncryptionInfo::ClevisInfo((pin, config)) => {
+                let mut parsed_config = config.clone();
+                let y = interpret_clevis_config(pin, &mut parsed_config)?;
+                Self::initialize_with_clevis(device, physical_path, (pin, &parsed_config, y))?
+            }
+        };
+
+        let activation_name = format_crypt_backstore_name(&pool_uuid);
+        activate(
+            device,
+            encryption_info.key_description(),
+            UnlockMethod::Any,
+            None,
+            &activation_name,
+        )
+    }
+
+    pub fn rollback(
+        device: &mut CryptDevice,
+        physical_path: &Path,
+        name: &DmName,
+    ) -> StratisResult<()> {
+        ensure_wiped(device, physical_path, name)
+    }
+
+    /// Acquire the crypt device handle for the physical path in this `CryptHandle`.
+    pub(super) fn acquire_crypt_device(&self) -> StratisResult<CryptDevice> {
+        acquire_crypt_device(self.luks2_device_path())
+    }
+
+    /// Query the device metadata to reconstruct a handle for performing operations
+    /// on an existing encrypted device.
+    ///
+    /// This method will check that the metadata on the given device is
+    /// for the LUKS2 format and that the LUKS2 metadata is formatted
+    /// properly as a Stratis encrypted device. If it is properly
+    /// formatted it will return the device identifiers (pool and device UUIDs).
+    ///
+    /// NOTE: This will not validate that the proper key is in the kernel
```

Comment:
I think in this implementation the ```can_unlock``` method no longer exists, so it makes sense to remove mention of it.

Reply:
Resolved.